### PR TITLE
A decision_proceduret isn't a messaget [blocks: #3800]

### PR DESCRIPTION
**Pull request by tautschnig**
Some derived classes may want to generate output, but then it's safe for them to
be messaget's.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: bfb6694c03cbd98f2e1da37afd16d3212c735430).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99535966
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by tautschnig** [comment]
I guess this needs a genuine update to TG as there are some API changes required. The message handler that may now be needed should be available wherever those functions are invoked.

File: src/goto-symex/symex_target_equation.cpp
Comment (peterschrammel) [inline src/goto-symex/symex_target_equation.cpp]: Can you add a `messaget log` member to reduce the churn?

Comment (tautschnig) [reply src/goto-symex/symex_target_equation.cpp]: I could do, but then the constructor would need to take a `message_handlert` (as constructing a `messaget` without a `message_handlert` is deprecated). Does that seem reasonable?

File: src/solvers/prop/prop_conv.h
Comment (romainbrenguier) [inline src/solvers/prop/prop_conv.h:71]: ideally the inheritance should be removed here as well (could be done in a separate PR)

**Review by romainbrenguier — COMMENTED** [review]
Needs a rebase. Otherwise looks good, but I'm waiting for tests result to give approval.

**Review by romainbrenguier — APPROVED** [review]

File: src/solvers/prop/prop_conv.h
Comment (tautschnig) [reply src/solvers/prop/prop_conv.h:71]: Yes, I am going to do so in a separate PR. I just wanted to keep the churn as small as possible.

File: src/goto-symex/symex_target_equation.cpp
Comment (tautschnig) [reply src/goto-symex/symex_target_equation.cpp]: @peterschrammel I have done as you suggested - I think it's thanks to your goto-checker refactoring that this was a fairly smooth sailing.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 4851c79904b2565eae1b19f293b52d683062a6ac).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100055608
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 3530bc85ba2554d96fd1560c61a5c7eaaee751a7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100982523
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by romainbrenguier** [comment]
Half of the CI checks on TG were passing when I made the pull request (the other half didn't start for some reason), I can't check it now because the build is not working for other reasons, but I wouldn't block this PR for that

**Comment by tautschnig** [comment]
Thanks @romainbrenguier! This is still in need of review anyway, but I'll remove the TG-approval label for now.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 46a89301996fcac8661eb6f3e352d91a6cefe378).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103243408
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

File: src/util/decision_procedure.h
Comment (allredj) [inline src/util/decision_procedure.h:16]: Not sure why these are needed.

**Review by allredj — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

File: src/util/decision_procedure.h
Comment (tautschnig) [reply src/util/decision_procedure.h:16]: `decision_procedure_text` returns a `std::string` and `print_assignment` takes a `std::ostream` as parameter. (Previously both includes were dragged in by `message.h`.)

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4d8662f85fed583125755cde6c5a36863a56b003).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103462478

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: ab09a6c134fc0e1555a8dd080fa95332942b946a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103491412

**Pull request merged**
